### PR TITLE
Infer schema if none exists

### PR DESCRIPTION
**Pull request by csweaver**
The schema file is intended to allow a user to specify types, display names, and metadata in the data file to exclude from the UI. This PR infers this from the anndata file so a scientist doesn't have to create an additional JSON file to see thier data.

### CODE REVIEW

File: server/app/scanpy_engine/scanpy_engine.py [hunk 1]
```diff
@@ -27,11 +27,38 @@ def _set_cell_names(self):
...
+                # Since there are many type of float/int in numpy datatypes the kind attribute of a datatype object
+                # offers a decent insight into whether it can be lumped in with floats or ints, which is what we
+                # care about here.
+                data_kind = self.data.obs[m].dtype.kind
+                variable_type = "categorical"
+                data_type = "string"
+                if data_kind == 'f':
+                    variable_type = "continuous"
+                    data_type = "float"
+                elif data_kind == 'i':
```
Comment (ttung) [inline server/app/scanpy_engine/scanpy_engine.py]: Do you want to capture unsigned ('u') as well?

Comment (csweaver) [reply server/app/scanpy_engine/scanpy_engine.py]: yes! thanks

File: server/app/scanpy_engine/scanpy_engine.py [hunk 2]
```diff
@@ -27,11 +27,38 @@ def _set_cell_names(self):
...
+                # Since there are many type of float/int in numpy datatypes the kind attribute of a datatype object
+                # offers a decent insight into whether it can be lumped in with floats or ints, which is what we
+                # care about here.
+                data_kind = self.data.obs[m].dtype.kind
+                variable_type = "categorical"
+                data_type = "string"
+                if data_kind == 'f':
+                    variable_type = "continuous"
+                    data_type = "float"
+                elif data_kind in ['i', 'u']:
```
Comment (ttung) [inline server/app/scanpy_engine/scanpy_engine.py]: nit: make it `('i', 'u')`

File: server/app/scanpy_engine/scanpy_engine.py [hunk 3]
```diff
@@ -27,11 +27,38 @@ def _set_cell_names(self):
...
+                    data_type = "int"
```
Comment (ttung) [inline server/app/scanpy_engine/scanpy_engine.py]: does this processing happen the same way if it's int or uint?

**Pull request merged**